packages/plugin-ext/src/common/proxy-handler.ts

```diff
@@ -25,11 +25,18 @@ export interface RpcHandlerOptions {
 }
 export interface ProxyHandlerOptions extends RpcHandlerOptions {
     channelProvider: () => Promise<Channel>,
+    proxySynchronizer: ProxySynchronizer,
 }
 
 export interface InvocationHandlerOptions extends RpcHandlerOptions {
     target: any
 }
+
+export interface ProxySynchronizer {
+    startProxyInitialization(id: string, init: Promise<void>): void
```

Comment:
Should we ease our lives and simply take a `Promise<RpcProtocol>`? Then we do not have to do an empty promise-chaining later.

Reply:
I would rather keep it this way since the `RpcProtocol` is private in the `ClientProxyHandler` and I don't see a benefit in handing it out.

Reply:
Fine with me.